libats-messaging/src/main/scala/com/advancedtelematic/libats/messaging/test/MockMessageBus.scala

```diff
@@ -0,0 +1,41 @@
+package com.advancedtelematic.libats.messaging.test
+
+import java.util.concurrent.ConcurrentHashMap
+
+import akka.http.scaladsl.util.FastFuture
+import com.advancedtelematic.libats.messaging.MessageBusPublisher
+import com.advancedtelematic.libats.messaging_datatype.MessageLike
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.reflect.ClassTag
+
+class MockMessageBus extends MessageBusPublisher {
+
+  private case class ReceivedMsg(tag: ClassTag[_], payload: Any)
+
+  private val received = new ConcurrentHashMap[String, ReceivedMsg]()
+
+  def wasReceived[T](filterFn: T => Boolean)(implicit msgLike: MessageLike[T]): Option[T] = {
+    val found: T = received.searchValues(1024, (msg: ReceivedMsg) => {
+      val maybeT = if(msg.tag.equals(msgLike.tag)) Some(msg.payload.asInstanceOf[T]) else None
+
+      maybeT match {
+        case Some(t) if filterFn(t) => t
+        case _ => null.asInstanceOf[T]
+      }
+    })
+
+    Option(found)
+  }
+
+  def wasReceived[T](id: String)(implicit msgLike: MessageLike[T]): Option[T] =
```

Comment:
Nitpicking here, but one would expect `wasReceived` to return a boolean. Maybe something like `find` or `search`?

Reply:
:+1: